File: src/Debt_LRA_Transcript/LectureRegistrationApplication.java
package Debt_LRA_Transcript;

import java.util.HashMap;
import java.util.Map;

import Enums.ApprovalState;
import lecture.LectureSession;
import person.Advisor;
import person.Student;

//Sena VATANSEVER 150119755

public class LectureRegistrationApplication {

	private Map<LectureSession, ApprovalState> sessionsSentForApproval;
	private Advisor advisor;
	private Student student;
	

	public LectureRegistrationApplication(Map<LectureSession, ApprovalState> sessionsSentForApproval, Advisor advisor,
			Student student) {
		super();
		this.sessionsSentForApproval = sessionsSentForApproval;
		this.advisor = advisor;
		this.student = student;
		
		if (this.sessionsSentForApproval == null) {
			this.sessionsSentForApproval = new HashMap<LectureSession, ApprovalState>();
		}
	}
	
	public Map<LectureSession, ApprovalState> getSessionsSentForApproval() {
		return this.sessionsSentForApproval;
	}

	public Student getStudent() {
		return student;
	}
	
	
}

File: pythonCode/LectureRegistrationApplication.py
class LectureRegistrationApplication():

    def __init__(self):
        pass

     # Creating properties for variables
    def getSessionsSentForApproval(self):
        try:
            x = self.__sessionsSentForApproval
        except AttributeError:
            return dict()
        return self.__sessionsSentForApproval

    def getAdvisor(self):
        return self.__advisor

    def getStudent(self):
        return self.__student

    def setSessionsSentForApproval(self, sessionsSentForApproval):
        self.__sessionsSentForApproval = sessionsSentForApproval

    def setStudent(self, student):
        self.__student = student

    def setAdvisor(self, advisor):
        self.__advisor = advisor
